compiler/rustc_const_eval/src/transform/check_consts/qualifs.rs

```diff
@@ -353,14 +353,15 @@ where
 
     // Check the qualifs of the value of `const` items.
     if let Some(ct) = constant.literal.const_for_ty() {
-        if let ty::ConstKind::Unevaluated(ty::Unevaluated { def, substs: _, promoted }) = ct.kind()
+        if let ty::ConstKind::Unevaluated(ty::Unevaluated { def, substs: _, promoted: _ }) =
+            ct.kind()
```

Comment:
this should only care about `ConstantKind::Unevaluated` afaict, never type system constants.

Reply:
We do need `ConstantKind::Ty` here, we get failing borrowck tests if we don't.